wire_test.go

```diff
@@ -206,6 +207,81 @@ func TestServerWritingResult(t *testing.T) {
 	})
 }
 
+func TestServerHandlingMultipleConnections(t *testing.T) {
+	handler := func(ctx context.Context, query string, writer DataWriter, parameters []string) error {
+		t.Log("serving query")
+		writer.Define(Columns{ //nolint:errcheck
+			{
+				Table:  0,
+				Name:   "age",
+				Oid:    oid.T_int4,
+				Width:  1,
+				Format: TextFormat,
+			},
+		})
+		writer.Row([]any{20}) //nolint:errcheck
+		return writer.Complete("OK")
+	}
+	d, err := zap.NewDevelopment()
+	require.NoError(t, err)
+	server, err := NewServer(SimpleQuery(handler), Logger(d))
+	require.NoError(t, err)
+	address := TListenAndServe(t, server)
+	databaseUrl := fmt.Sprintf("postgres://%s:%d", address.IP, address.Port)
+	conn, err := sql.Open("pgx", databaseUrl)
+	require.NoError(t, err)
+	t.Cleanup(func() {
+		_ = conn.Close()
+	})
+
+	t.Run("test connectivity", func(t *testing.T) {
+		err := conn.Ping()
+		require.NoError(t, err)
+	})
+
+	t.Run("simple query execution", func(t *testing.T) {
+		rows, err := conn.Query("select age from person")
+		require.NoError(t, err)
+		t.Cleanup(func() {
+			rows.Close()
+		})
+		require.True(t, rows.Next())
+		require.NoError(t, rows.Err())
+	})
+
+	t.Run("prepared statement query execution using dollar parm style", func(t *testing.T) {
+		query := "select age from person where age > $1"
+		stmt, err := conn.Prepare(query)
+		require.NoError(t, err)
+		t.Cleanup(func() {
+			stmt.Close()
+		})
+		rows, err := stmt.Query(1)
+		require.NoError(t, err)
+		t.Cleanup(func() {
+			rows.Close()
+		})
+		require.True(t, rows.Next())
+		require.NoError(t, rows.Err())
+	})
```

Comment:
These sub tests are part of the test above. I would like to suggest to not place them inside of a sub test since they cannot be executed individually. Feel free to create some wrapper/mock functions in order to simplify the test and reduce code duplicates.

Reply:
Not fully getting what you mean, if you see below in my IDE (IntelliJ) i am able to execute the subtests individually as well (screenshot below). Yeah i do agree that the code repetition in these sub tests can be reduced

<img width="800" alt="Screenshot 2023-01-02 at 12 34 02 PM" src="https://user-images.githubusercontent.com/763760/210202470-85f32156-5298-4e1f-bb26-7c0b0808d8fb.png">

Reply:
Basically the above fancy UI runs the following command to just run the subtest

```go
go test -run '^\QTestServerHandlingMultipleConnections\E$/^\Qsimple_query_execution\E$'
```

Reply:
Have refactored the above to reduce the amount of duplication in the prepared statement scenarios.